src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -138,21 +139,93 @@ impl Default for WorkunitMetadata {
   }
 }
 
-#[derive(Clone, Default)]
+type CompletedWorkunitReceiver =
+  Arc<Mutex<Receiver<(SpanId, Option<WorkunitMetadata>, SystemTime)>>>;
+type CompletedWorkunitSender = Arc<Mutex<Sender<(SpanId, Option<WorkunitMetadata>, SystemTime)>>>;
+
+#[derive(Clone)]
 pub struct WorkunitStore {
   rendering_dynamic_ui: bool,
+  streaming_workunit_data: StreamingWorkunitData,
+  heavy_hitters_data: HeavyHittersData,
+}
+
+#[derive(Clone)]
+struct StreamingWorkunitData {
+  started_workunits_rx: Arc<Mutex<Receiver<Workunit>>>,
+  started_workunits_tx: Arc<Mutex<Sender<Workunit>>>,
+  completed_workunits_rx: CompletedWorkunitReceiver,
+  completed_workunits_tx: CompletedWorkunitSender,
```

Comment:
It might be slightly easier/more-efficient to have one channel containing both "start" and "finish" events as an enum?